### ISSUE TITLE
`golangci-lint run` doesn't show issues from explicitly defined formatters

### ISSUE THREAD

**Issue by zchenyu** [golangci/golangci-lint#6340]
### Welcome

- [x] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/docs/welcome/faq/#why-do-you-have-typecheck-errors).
- [x] Yes, I've tried with the standalone [linter](https://golangci-lint.run/docs/linters/) if available (e.g., gocritic, go vet, etc.).
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

Official binary

### Description of the problem

Running `golangci-lint run` doesn't execute the formatters explicitly defined in `.golangci`. According to the [docs](https://golangci-lint.run/docs/configuration/cli/) and [this issue](https://github.com/golangci/golangci-lint-action/issues/1304#issuecomment-3494308225) indicates that it should report formatter issues.

e.g.
```
# main.go has an extra leading \t
$ cat main.go
	package main
```

```
# gofmt behaves as expected
$ gofmt -s main.go 
package main

# golangci-lint fmt behaves as expected
$ golangci-lint fmt --diff 
diff main.go.orig main.go
--- main.go.orig
+++ main.go
@@ -1,2 +1,1 @@
-	package main
+package main

# golangci-lint run doesn't show any issues
$ golangci-lint run
0 issues.
```

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 2.8.0 built with go1.25.5 from e2e40021 on 2026-01-07T21:29:47Z
```

</details>

### Configuration

<details>

```console
version: "2"
formatters:
  enable:
    - gofmt
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.25.0 linux/amd64
AR='ar'
CC='gcc'
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_ENABLED='0'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
CXX='g++'
GCCGO='gccgo'
GO111MODULE=''
GOAMD64='v1'
GOARCH='amd64'
GOAUTH='netrc'
GOBIN=''
GOCACHE='/home/zchenyu/.cache/go-build'
GOCACHEPROG=''
GODEBUG=''
GOENV='/home/zchenyu/.config/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFIPS140='off'
GOFLAGS=''
GOGCCFLAGS='-fPIC -m64 -Wl,--no-gc-sections -fmessage-length=0 -ffile-prefix-map=/tmp/go-build4280099624=/tmp/go-build -gno-record-gcc-switches'
GOHOSTARCH='amd64'
GOHOSTOS='linux'
GOINSECURE=''
GOMOD='/home/zchenyu/tmp/golangci/go.mod'
GOMODCACHE='/home/zchenyu/go/pkg/mod'
GONOPROXY='github.com/fw-ai'
GONOSUMDB='github.com/fw-ai'
GOOS='linux'
GOPATH='/home/zchenyu/go'
GOPRIVATE='github.com/fw-ai'
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/usr/local/go'
GOSUMDB='sum.golang.org'
GOTELEMETRY='local'
GOTELEMETRYDIR='/home/zchenyu/.config/go/telemetry'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/usr/local/go/pkg/tool/linux_amd64'
GOVCS=''
GOVERSION='go1.25.0'
GOWORK=''
PKG_CONFIG='pkg-config'
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO golangci-lint has version 2.8.0 built with go1.25.5 from e2e40021 on 2026-01-07T21:29:47Z 
INFO [config_reader] Config search paths: [./ /home/zchenyu/tmp/golangci /home/zchenyu/tmp /home/zchenyu /home /] 
INFO [config_reader] Module name "example.com"    
INFO maxprocs: Leaving GOMAXPROCS=16: CPU quota undefined 
INFO [goenv] Read go env for 4.451971ms: map[string]string{"GOCACHE":"/home/zchenyu/.cache/go-build", "GOROOT":"/usr/local/go"} 
INFO [lintersdb] Active 5 linters: [errcheck govet ineffassign staticcheck unused] 
INFO [loader] Go packages loading at mode 8767 (exports_file|name|compiled_files|deps|files|imports|types_sizes) took 54.195445ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 93.226µs 
INFO [linters_context/goanalysis] analyzers took 6.906249ms with top 10 stages: buildir: 1.034783ms, fact_deprecated: 428.739µs, printf: 385.648µs, SA5012: 269.93µs, ctrlflow: 265.571µs, fact_purity: 264.93µs, nilness: 254.33µs, typedness: 237.919µs, S1030: 164.811µs, errcheck: 152.347µs 
INFO [runner] processing took 3.469µs with stages: max_same_issues: 551ns, exclusion_paths: 471ns, exclusion_rules: 441ns, nolint_filter: 331ns, source_code: 281ns, path_absoluter: 271ns, path_shortener: 211ns, max_from_linter: 201ns, path_prettifier: 201ns, severity-rules: 150ns, max_per_file_from_linter: 60ns, filename_unadjuster: 40ns, cgo: 40ns, diff: 40ns, uniq_by_line: 30ns, sort_results: 30ns, generated_file_filter: 30ns, path_relativity: 30ns, fixer: 30ns, invalid_issue: 30ns 
INFO [runner] linters took 6.272637ms with stages: goanalysis_metalinter: 6.229255ms 
0 issues.
INFO File cache stats: 0 entries of total size 0B 
INFO Memory: 2 samples, avg is 40.2MB, max is 44.6MB 
INFO Execution took 65.458019ms  
```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
	package main
```

</details>

### Validation

- [x] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a sponsor/backer through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by zchenyu** [comment]
I'm stupid. I named it `.golintci.yml` instead of `.golangci.yml`.
